src/pykeen/evaluation/rank_based_evaluator.py

```diff
@@ -473,6 +480,127 @@ def process_scores_(
+
+        evaluator = _OGBEvaluatorBridge()
+        # this setting is equivalent to the WikiKG2 setting, and will calculate MRR *and* H@k for k in {1, 3, 10}
+        evaluator.eval_metric = "mrr"
+        evaluator.K = None
+
+        # filter supported metrics
+        metrics: List[RankBasedMetric] = []
+        for metric in self.metrics:
+            if not isinstance(metric, (HitsAtK, InverseHarmonicMeanRank)) or (
+                isinstance(metric, HitsAtK) and metric.k not in {1, 3, 10}
+            ):
+                logger.warning(f"{metric} is not supported by OGB evaluator")
+                continue
+            metrics.append(metric)
+
+        # prepare input format, cf. `evaluator.expected_input``
+        # y_pred_pos: shape: (num_edge,)
+        # y_pred_neg: shape: (num_edge, num_nodes_neg)
+        y_pred_pos: Dict[Target, torch.Tensor] = {}
+        y_pred_neg: Dict[Target, torch.Tensor] = {}
+
+        num_triples = mapped_triples.shape[0]
+        device = mapped_triples.device
+        # iterate over prediction targets
+        for target, negatives in self.negative_samples.items():
+            # pre-allocate
+            # TODO: maybe we want to collect scores on CPU / add an option?
+            y_pred_pos[target] = y_pred_pos_side = torch.empty(size=(num_triples,), device=device)
+            num_negatives = negatives.shape[1]
+            y_pred_neg[target] = y_pred_neg_side = torch.empty(size=(num_triples, num_negatives), device=device)
+            # iterate over batches
+            offset = 0
+            for hrt_batch, negatives_batch in zip(
+                mapped_triples.split(split_size=batch_size), negatives.split(split_size=batch_size)
+            ):
+                # combine ids, shape: (batch_size, num_negatives + 1)
+                ids = torch.cat([hrt_batch[:, 2, None], negatives_batch], dim=1)
+                # get scores, shape: (batch_size, num_negatives + 1)
+                scores = model.predict(hrt_batch=hrt_batch, target=target, ids=ids, mode=self.mode, **kwargs)
+                # store positive and negative scores
+                this_batch_size = scores.shape[0]
+                stop = offset + this_batch_size
+                y_pred_pos_side[offset:stop] = scores[:, 0]
+                y_pred_neg_side[offset:stop] = scores[:, 1:]
+                offset = stop
+
+        def iter_preds() -> Iterable[Tuple[ExtendedTarget, torch.Tensor, torch.Tensor]]:
+            """Iterate over predicted scores for extended prediction targets."""
+            targets = sorted(y_pred_pos.keys())
+            for target in targets:
+                yield target, y_pred_pos[target], y_pred_neg[target]
+            yield SIDE_BOTH, torch.cat([y_pred_pos[t] for t in targets], dim=0), torch.cat(
+                [y_pred_neg[t] for t in targets], dim=0
+            )
+
+        result: Dict[Tuple[str, ExtendedTarget, RankType], float] = {}
+        # TODO: ogb's rank-type is non-deterministic...
```

Comment:
Question is now that they accepted This Pr when they will make a new release

Reply:
cf. https://github.com/snap-stanford/ogb/pull/357#issuecomment-1214878757